### PR TITLE
Add a FuzzNoCustom() function

### LINKED ISSUE #10: Idea: Have a way to say "fuzz all fields except.."
In my custom fuzzers I sometimes want to fuzz all fields of a struct except a few that I do manually.

### PR DESCRIPTION
**Pull request by thockin**
This skips custom fuzz functions.  The use case is for custom-fuzzing a subset
of fields of a larger struct.  The custom function can call this on the struct
(to avoid a recursive call) and then patch-up fields it needs to handle
specially.

Better solution to #10.

Fixes #10

### CODE REVIEW

File: fuzz.go [hunk 1]
```diff
@@ -146,20 +146,41 @@ func (f *Fuzzer) Fuzz(obj interface{}) {
...
+	if v.Kind() != reflect.Ptr {
+		panic("needed ptr!")
 	}
-	// Check for both pointer and non-pointer custom functions.
-	if v.CanAddr() && f.tryCustom(v.Addr()) {
+	v = v.Elem()
+	f.doFuzz(v, flagNoCustomFuzz)
+}
+
+const flagNoCustomFuzz uint64 = 0x01 // Do not try to find a custom fuzz function.
```
Comment (lavalamp) [inline fuzz.go]:
```go
const (
  // Do not try to find a custom fuzz function. Does not apply recursively.
  flagNoCustomFuzz uint64 = 1 << iota
)
```

File: fuzz.go [hunk 2]
```diff
@@ -146,20 +146,41 @@ func (f *Fuzzer) Fuzz(obj interface{}) {
...
-func (f *Fuzzer) doFuzz(v reflect.Value) {
-	if !v.CanSet() {
-		return
+// FuzzNoCustom is just like Fuzz, except that any custom fuzz function for
+// obj's type will not be called.  This applies only to obj and not other
+// instances of obj's type.
+// Not safe for cyclic or tree-like structs!
+// obj must be a pointer. Only exported (public) fields can be set (thanks, golang :/ )
+// Intended for tests, so will panic on bad input or unimplemented fields.
+func (f *Fuzzer) FuzzNoCustom(obj interface{}) {
```
Comment (lavalamp) [inline fuzz.go]: What about the name DefaultFuzz?

**Comment by lavalamp** [comment]
LGTM - minor nits

**Comment by lavalamp** [comment]
Wait-- actually, can you add a test? :)

File: fuzz.go (hunk 1 above)
Comment (thockin) [inline fuzz.go]: Done in next push

File: fuzz.go (hunk 2 above)
Comment (thockin) [inline fuzz.go]: It's totally your call.  I like that it started with Fuzz, but I'll go with whatever you like :)

**Comment by thockin** [comment]
There is a test at the end of fuzz_test.go

**Comment by thockin** [comment]
New push is up

**Comment by lavalamp** [comment]
Sorry to be both a pain and unclear-- a test for *Continue's* FuzzNoCustom would be great, since that's probably how most people will consume this feature.

File: fuzz.go (hunk 2 above)
Comment (lavalamp) [inline fuzz.go]: That's fine, I don't have a strong opinion.

**Comment by thockin** [comment]
Ah, right.  Now calls {Fuzzer,Continue}.{Fuzz,FuzzNoCustom}() - PTAL

**Comment by thockin** [comment]
And once this is in we can update to it in kubernetes, yes?

**Comment by lavalamp** [comment]
LGTM, thanks!

**Comment by lavalamp** [comment]
Yeah, should be easy to update it in k8s.

**Pull request merged**